src/main/java/net/objecthunter/exp4j/shuntingyard/ShuntingYard.java

```diff
@@ -51,8 +51,11 @@
                 output.add(token);
                 break;
             case Token.TOKEN_FUNCTION:
-                stack.add(token);
-                break;
+				if(!stack.empty() && stack.peek().getType() == Token.TOKEN_FUNCTION) {
```

Comment:
Hi @alexinul,

Consider adding test case for this in https://github.com/fasseg/exp4j/blob/master/src/test/java/net/objecthunter/exp4j/shuntingyard/ShuntingYardTest.java, for example:

```java
@Test(expect = IllegalArgumentException.class)
public void testShuntingYardMultiFunctionConcat() throws Exception {
  ShuntingYard.convertToRPN("sincos(x)", null, null, null, true);
}
```